File: sdk/android/src/main/java/com/tuya/smart/rnsdk/utils/ReactParamsCheck.java
package com.tuya.smart.rnsdk.utils;

import com.facebook.react.bridge.ReadableMap;

import java.util.List;

public class ReactParamsCheck {

    public static Boolean checkParams(List<String> keys, ReadableMap params) {
        for (int i = 0; i < keys.size(); i++) {
            if (!params.hasKey(keys.get(i))) {
                throw new IllegalArgumentException("need key"+keys.get(i));
            }
        }
        return true;
    }
}

File: android/src/main/java/com/tuya/smart/rnsdk/utils/ReactParamsCheck.kt
package com.tuya.smart.rnsdk.utils

import com.facebook.react.bridge.ReadableMap

object ReactParamsCheck {
    /**
     *  接口输入参数check
     */
    fun checkParams(keys: Array<String>, params: ReadableMap): Boolean {
        for (key in keys) {
            if (!params.hasKey(key)) {
                throw  IllegalArgumentException("need ${key}")
            }
        }
        return true
    }
}